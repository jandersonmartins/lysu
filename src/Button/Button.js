import * as S from './Button.styles'

export default S.Button
