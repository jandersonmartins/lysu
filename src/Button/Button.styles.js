import styled from 'styled-components'

const Button = styled.button`
  background-color: ${props => props.theme.colors.primary};
  color: ${props => props.theme.colors.white};
  padding: .6rem 1.2rem;
  border: .1rem solid ${props => props.theme.colors.primaryDark};
  border-radius: .3rem;
  font-weight: 700;
  cursor: pointer;

  &:disabled {
    opacity: .1;
    cursor: not-allowed;
  }
`

export { Button }
