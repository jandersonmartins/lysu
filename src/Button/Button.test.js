import React from 'react'
import { render } from '@testing-library/react'
import { ThemeProvider } from 'styled-components'
import Button from './Button'

test('renders correctly', () => {
  const { container } = render(
    <ThemeProvider theme={{ colors: {} }}>
      <Button>Click</Button>
    </ThemeProvider>
  )

  expect(container.firstChild).toMatchSnapshot()
})
