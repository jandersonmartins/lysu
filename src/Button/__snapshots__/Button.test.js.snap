// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
.c0 {
  padding: .6rem 1.2rem;
  border: .1rem solid;
  border-radius: .3rem;
  font-weight: 700;
  cursor: pointer;
}

.c0:disabled {
  opacity: .1;
  cursor: not-allowed;
}

<button
  class="c0"
>
  Click
</button>
`;
